java/vector/src/test/java/org/apache/arrow/vector/util/DecimalUtilityTest.java

```diff
@@ -32,6 +32,31 @@
   private static final BigInteger[] MIN_BIG_INT = new BigInteger[]{MAX_BIG_INT[0].multiply(BigInteger.valueOf(-1)),
      MAX_BIG_INT[1].multiply(BigInteger.valueOf(-1))};
 
+  @Test
+  public void testSetLongInDecimalArrowBuf() {
+    try (BufferAllocator allocator = new RootAllocator(128);
+         ArrowBuf buf = allocator.buffer(16);
+    ) {
+      int [] intValues = new int [] {Integer.MAX_VALUE, Integer.MIN_VALUE, 0};
+      for (int val : intValues) {
+        buf.clear();
+        DecimalUtility.writeLongToArrowBuf((long) val, buf, 0);
+        BigDecimal actual = DecimalUtility.getBigDecimalFromArrowBuf(buf, 0, 0);
+        BigDecimal expected = BigDecimal.valueOf(val);
+        Assert.assertEquals(expected, actual);
+      }
+
+      long [] longValues = new long[] {Long.MIN_VALUE, 0 , Long.MAX_VALUE};
```

Comment:
I don't think you need 2 loops here, just move the `Integer.MAX_VALUE` and `MIN_VALUE` here

Reply:
Sure, I will drop lines 49-57.